main.tf

```diff
@@ -13,38 +15,44 @@ resource "aws_security_group" "this" {
 }
 
 resource "aws_security_group_rule" "egress" {
-  description       = var.egress_security_group_rule_description
+  count = var.create ? 1 : 0
+
   cidr_blocks       = ["0.0.0.0/0"] #tfsec:ignore:AWS007
+  description       = var.egress_security_group_rule_description
   from_port         = 0
   protocol          = "-1"
-  security_group_id = aws_security_group.this.id
+  security_group_id = aws_security_group.this[0].id
   to_port           = 0
   type              = "egress"
 }
 
 resource "aws_security_group_rule" "ingress_cidr" {
-  count             = local.create_ingress_cidr_sg_rule ? 1 : 0
+  count = local.create_ingress_cidr_sg_rule ? 1 : 0
+
   cidr_blocks       = var.ingress_rule_cidr_blocks
   description       = var.ingress_cidr_sg_rule_description
-  from_port         = var.port
+  from_port         = local.port
   protocol          = "tcp"
-  security_group_id = aws_security_group.this.id
-  to_port           = var.port
+  security_group_id = aws_security_group.this[0].id
+  to_port           = local.port
   type              = "ingress"
 }
 
 resource "aws_security_group_rule" "ingress_sg" {
-  source_security_group_id = each.value
+  count = length(var.security_group_ids)
+
   description              = var.ingress_sg_sg_rule_description
-  for_each                 = var.security_group_ids
-  from_port                = var.port
+  from_port                = local.port
   protocol                 = "tcp"
-  security_group_id        = aws_security_group.this.id
-  to_port                  = var.port
+  security_group_id        = aws_security_group.this[0].id
+  source_security_group_id = var.security_group_ids[count.index]
+  to_port                  = local.port
```

Comment:
btw there is a var.port line 90 or 96 in the variables.tf

Reply:
this is what happens when you're working on 7 PRs for 6 modules in one day. 😞